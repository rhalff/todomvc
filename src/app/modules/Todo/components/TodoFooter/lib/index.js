// @flow
export { default as hasCompletedItems } from './hasCompletedItems'
export { default as getUncompletedItemCount } from './getUncompletedItemCount'
