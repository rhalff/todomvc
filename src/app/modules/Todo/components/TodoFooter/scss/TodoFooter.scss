@import 'colors';

.todoFooter {
  display: flex;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;

  color: $todoFooterColor;
  padding: 10px 15px;
  height: 20px;
  text-align: center;
  border-top: 1px solid $todoFooterBorderTopColor;

  @media (max-width: 430px) {
    height: 50px;
  }

  > {
    :first-child {
      flex: 0 0 auto;
    }
    :nth-child(2) {
      flex: 1 1 auto;
    }
    :last-child {
      flex: 0 0 auto;
    }
  }
}

.todoFooter:before {
  content: '';
  position: absolute;
  right: 0;
  bottom: 0;
  left: 0;
  height: 50px;
  overflow: hidden;
  box-shadow: 0 1px 1px rgba(0, 0, 0, 0.2), 0 8px 0 -3px #f6f6f6, 0 9px 1px -3px rgba(0, 0, 0, 0.2), 0 16px 0 -6px #f6f6f6, 0 17px 2px -6px rgba(0, 0, 0, 0.2);
  pointer-events: none;
}
