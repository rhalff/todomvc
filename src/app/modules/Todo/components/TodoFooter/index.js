// @flow
import React from 'react'
import styles from './scss/TodoFooter.scss'
import ItemCounter from '../ItemCounter'
import ButtonBar from '../ButtonBar'
import ClearCompletedButton from '../ClearCompletedButton'
import { getUncompletedItemCount, hasCompletedItems } from './lib'

import type { Todos } from '../../types/todos'
import type { Buttons } from '../../types/buttons'

const filterList: Buttons = [
  {
    name: 'ALL',
    title: 'All'
  },
  {
    name: 'ACTIVE',
    title: 'Active'
  },
  {
    name: 'COMPLETED',
    title: 'Completed'
  }
]

type Props = {
  filter: string,
  items: Todos,
  onClearCompleted: Function,
  onFilterClick: Function
}

const TodoFooter = ({
  filter,
  items,
  onClearCompleted,
  onFilterClick
}: Props) => (
  <footer className={styles.todoFooter}>
    <ItemCounter itemCount={getUncompletedItemCount(items)} />
    <ButtonBar buttons={filterList} filter={filter} onClick={onFilterClick} />
    {hasCompletedItems(items) && (
      <ClearCompletedButton onClear={onClearCompleted} />
    )}
  </footer>
)

export default TodoFooter
