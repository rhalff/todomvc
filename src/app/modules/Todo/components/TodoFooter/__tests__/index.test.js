// @flow
import React from 'react'
import TodoFooter from '../'
import renderer from 'react-test-renderer'

const todos = [{ id: 1, text: 'Test Todo' }, { id: 2, text: 'Test Todo 2' }]

test('Can show Todo Footer', () => {
  const component = renderer.create(
    <TodoFooter
      items={todos}
      onFilterClick={() => {}}
      onClearCompleted={() => {}}
    />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
