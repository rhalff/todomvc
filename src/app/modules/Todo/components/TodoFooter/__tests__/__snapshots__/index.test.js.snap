// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show Todo Footer 1`] = `
<footer
  className="todoFooter"
>
  <div
    className="itemCounter"
  >
    <span>
      2
       
      items
       left
    </span>
  </div>
  <ul
    className="buttonBar"
  >
    <li
      className="listItem"
      onClick={[Function]}
    >
      <a
        className="filterButton"
        href="/#ALL"
        name="ALL"
      >
        All
      </a>
    </li>
    <li
      className="listItem"
      onClick={[Function]}
    >
      <a
        className="filterButton"
        href="/#ACTIVE"
        name="ACTIVE"
      >
        Active
      </a>
    </li>
    <li
      className="listItem"
      onClick={[Function]}
    >
      <a
        className="filterButton"
        href="/#COMPLETED"
        name="COMPLETED"
      >
        Completed
      </a>
    </li>
  </ul>
</footer>
`;
